OSDOCS-5637-bugbatch2: Documented another batch of 4.13 z-stream notes (#59456)
openshift/openshift-docs: release_notes/ocp-4-13-release-notes.adoc

* The previous resolution of link:https://issues.redhat.com/browse/OCPBUGS-5546[OCPBUGS-5546] removed the `clusterName` assignment of `MachineConfig.Name` in the machine configuration object. As a result, the value of the parameter was an empty string and, when it was combined with the value of `machineName` to create an IP address name, it created an invalid value. The invalid value caused machines to fail during provisioning. With this release, the value for `clusterName` is obtained from the infrastructure object so that it creates a valid IP address name. (link:https://issues.redhat.com/browse/OCPBUGS-7696[*OCPBUGS-7696*])

> * The Kubernetes 1.26 release introduced changes to the node infrastructure, such as removing an unhealthy node with an `NotReady` status from the public load balancer to prevent the node from receiving routing traffic. These changes can impact a {product-title} node that runs inside a cluster on Microsoft Azure, because removing a node from a public load balancer can result in the node losing internet connectivity. This issue might render the node unable to regain a ready status and establish an outbound connection. The {product-title} {product-version} release fixes this issue  so that a node marked with `NotReady` status is detected by the `kube-proxy` based health probes, now the default for services of `type=LoadBalancer` with `externalTrafficPolicy=Cluster`, without the need of node detachment from the public load balancer. This means that a node can retain an outbound internet connection throughout these phases. (link:https://issues.redhat.com/browse/OCPBUGS-7359[*OCPBUGS-7359*])

Reviewer: ```suggestion
* The Kubernetes 1.26 release introduced changes to the node infrastructure, such as removing an unhealthy node with an `NotReady` status from the public load balancer to prevent the node from receiving routing traffic. These changes impacted a node that ran inside a cluster on Microsoft Azure. As a result, the node was unable to regain a ready status and establish an outbound connection. With this update, a node marked with `NotReady` status is detected by the `kube-proxy` based health probes without the need of node detachment from the public load balancer. This means that a node can retain an outbound internet connection throughout these phases. (link:https://issues.redhat.com/browse/OCPBUGS-7359[*OCPBUGS-7359*])
```

[discrete]
[id="ocp-4-13-cloud-cred-operator-bug-fixes"]
==== Cloud Credential Operator

> * Amazon S3 changed their Amazon S3 bucket configuration, so that a bucket created in a Amazon Web Services (AWS) region has S3 Block Public Access enabled and access control limits (ACLs) disabled by default. This configuration limits S3 bucket resources to private use. The {product-title} (product-version} updates the CCO utility (`ccoctl`) to account for the default S3 bucket configuration so that S3 bucket resources are publicly available. (link:https://issues.redhat.com/browse/OCPBUGS-11706[*OCPBUGS-11706*])

Reviewer: ```suggestion
* Previously, Amazon Simple Storage Service (Amazon S3) changed their Amazon S3 bucket configuration, so a bucket created in a Amazon Web Services (AWS) region has S3 Block Public Access enabled and access control limits (ACLs) disabled by default. This configuration limits S3 bucket resources to private use. {product-title} (product-version} updates the CCO utility (`ccoctl`) to account for the default S3 bucket configuration so that S3 bucket resources are publicly available. (link:https://issues.redhat.com/browse/OCPBUGS-11706[*OCPBUGS-11706*])
```

Reviewer: Okay it might be better to do 

> Amazon Simple Storage Service (Amazon S3) updated the Amazon S3 bucket configuration so a bucket created in a Amazon Web Services (AWS) region has S3 Block Public Access enabled and access control limits (ACLs) disabled by default.

Just a thought

Author: Updated.

* Previously, when editing any pipeline in the {product-title} console, the correct data was not rendered in the *Pipeline builder* and *YAML view* configuration options. Because of this issue, you could not edit the pipeline in the *Pipeline builder*. With this update, data is parsed correctly and you can edit the pipeline using the builder. (link:https://issues.redhat.com/browse/OCPBUGS-5016[*OCPBUGS-5016*])

[discrete]
[id="ocp-4-13-cloud-etcd-operator-bug-fixes"]
==== etcd Cluster Operator

* Previously, the `ControlPlaneMachineSet` Operator attempted to recreate a control plane machine before the cluster bootstrapping completed. This resulted in the removal of the bootstrap node from the etcd cluster membership that caused etcd quorum loss and the cluster to go offline. With this update, `ControlPlaneMachineSet` Operator only recreates a control plane machine after the etcd Cluster Operator removes the bootstrap node. (link:https://issues.redhat.com/browse/OCPBUGS-10960[*OCPBUGS-10960*]

[discrete]
[id="ocp-4-13-hypershift-bug-fixes"]

> ==== HyperShift

Reviewer: Just one more nit I didn't notice before. Sorry @dfitzmau. I think this should be Hosted Control Planes @lahinson can you confirm?

Reviewer: I did some digging and I think hosted control planes is what we should go with for it based on https://docs.openshift.com/container-platform/4.12/release_notes/ocp-4-12-release-notes.html#ocp-4-12-hcp

Author: Thanks, updated.

* Previously, when editing any pipeline in the {product-title} console, the correct data was not rendered in the *Pipeline builder* and *YAML view* configuration options. Because of this issue, you could not edit the pipeline in the *Pipeline builder*. With this update, data is parsed correctly and you can edit the pipeline using the builder. (link:https://issues.redhat.com/browse/OCPBUGS-5016[*OCPBUGS-5016*])

[discrete]
[id="ocp-4-13-cloud-etcd-operator-bug-fixes"]
==== etcd Cluster Operator

> * Previously, the `ControlPlaneMachineSet` Operator attempted to recreate a master machine before the cluster bootstrapping completed. This issue could result in the removal of the bootstrap node from the etcd cluster membership that then caused etcd quorum loss and the cluster going offline. With this update, `ControlPlaneMachineSet` Operator only recreates a master machine after the etcd Cluster Operator removes the bootstrap node. (link:https://issues.redhat.com/browse/OCPBUGS-10960[*OCPBUGS-10960*]

Reviewer: ```suggestion
* Previously, the `ControlPlaneMachineSet` Operator attempted to recreate a master machine before the cluster bootstrapping completed. This resulted in the removal of the bootstrap node from the etcd cluster membership that caused etcd quorum loss and the cluster to go offline. With this update, `ControlPlaneMachineSet` Operator only recreates a master machine after the etcd Cluster Operator removes the bootstrap node. (link:https://issues.redhat.com/browse/OCPBUGS-10960[*OCPBUGS-10960*]
```

> * Previously, the `ControlPlaneMachineSet` Operator attempted to recreate a master machine before the cluster bootstrapping completed. This issue could result in the removal of the bootstrap node from the etcd cluster membership that then caused etcd quorum loss and the cluster going offline. With this update, `ControlPlaneMachineSet` Operator only recreates a master machine after the etcd Cluster Operator removes the bootstrap node. (link:https://issues.redhat.com/browse/OCPBUGS-10960[*OCPBUGS-10960*]

Reviewer: Jeana is out I think. Andrea might be able to help with this one.

> * Previously, the `ControlPlaneMachineSet` Operator attempted to recreate a master machine before the cluster bootstrapping completed. This issue could result in the removal of the bootstrap node from the etcd cluster membership that then caused etcd quorum loss and the cluster going offline. With this update, `ControlPlaneMachineSet` Operator only recreates a master machine after the etcd Cluster Operator removes the bootstrap node. (link:https://issues.redhat.com/browse/OCPBUGS-10960[*OCPBUGS-10960*]

Author: Thanks. I posted a message in our [Slack](https://redhat-internal.slack.com/archives/C85JYPHL3/p1683278039753219) group.

> * Previously, the `ControlPlaneMachineSet` Operator attempted to recreate a master machine before the cluster bootstrapping completed. This issue could result in the removal of the bootstrap node from the etcd cluster membership that then caused etcd quorum loss and the cluster going offline. With this update, `ControlPlaneMachineSet` Operator only recreates a master machine after the etcd Cluster Operator removes the bootstrap node. (link:https://issues.redhat.com/browse/OCPBUGS-10960[*OCPBUGS-10960*]

Reviewer 2: Catching up on emails, should be "control plane machine" (looks like that's where you landed :slightly_smiling_face:). Also "Control Plane Machine Set Operator", FYI.

* Previously, when downloading a `kubeconfig` file for a ServiceAccount, an error was displayed and the ServiceAccount token was unable to be reached. This error was due to the removal of automatically generated secrets. With this update, the download `kubeconfig` action has been removed and the error no longer occurs. (link:https://issues.redhat.com/browse/OCPBUGS-7308[*BZ#OCPBUGS-7308*])

> * Previously, if one or more existing plugins failed, the web console would not display a toast notification after an operator added a new plugin to the web console. This means that you would not receive a toast notification that prompts you to refresh the web console to view the new plugin. With this update, the web console checks when you added a plugin and then displays a toast notification on the console, regardless of any previously failed plugins. (link:https://issues.redhat.com/browse/OCPBUGS-10249[*OCPBUGS-10249*])

Reviewer: I think you could just remove the second sentence in this.
```suggestion
* Previously, if one or more existing plugins failed, the web console would not display a toast notification after an operator added a new plugin.  With this update, the web console checks when you added a plugin and then displays a toast notification on the console, regardless of any previously failed plugins. (link:https://issues.redhat.com/browse/OCPBUGS-10249[*OCPBUGS-10249*])
```

Author: It's important to mention the reason for the notification in the context of this note. I did not remove that sentence.

Reviewer: Another suggestion. Just thinking we should remove the "this means"

> Previously, if one or more existing plugins failed, the web console would not display a toast notification that prompts you to refresh the web console to view the new plugin after an operator added a new plugin to the web console.  With this update, the web console checks when you added a plugin and then displays a toast notification on the console, regardless of any previously failed plugins. (link:https://issues.redhat.com/browse/OCPBUGS-10249[*OCPBUGS-10249*])

Author: Thanks for the suggestion. The first sentence is over [32 words](https://www.ibm.com/docs/en/ibm-style?topic=medium-conversational-style) long. Restructured based on your example. 

How about:

Previously, if one or more existing plugins failed, the web console would not display a toast notification that prompted you to refresh the console. This action is required so that you can view a plugin after an operator adds the plugin to the console. With this update, the web console checks when the operator adds a plugin and then displays a toast notification on the console, regardless of any previously failed plugins. (link:https://issues.redhat.com/browse/OCPBUGS-10249[*OCPBUGS-10249*])

* Operator Lifecycle Manager (OLM) manages a set of `CatalogSource` objects from which Operators can be searched for and installed from. These catalog sources are the default sources for this action and are managed by Red Hat. However, it was possible to change these default catalog sources in a way that the OLM system would not notice. Modifying a default catalog source in a way that rendered it inoperable could cause cascading issues through OLM that might prevent a user from installing new or upgrading existing Operators on their cluster. This bug fix updates the `catalog-operator` runtime, which manages the default catalog sources, to be made aware of other changes to the `CatalogSource` spec. As a result, when a change is made to a default catalog source, OLM detects the change and resets it back to default. (link:https://issues.redhat.com/browse/OCPBUGS-5466[*OCPBUGS-5466*])

[discrete]
[id="ocp-4-13-perf-addon-operator-bug-fixes"]
==== Performance Addon Operator

> * Previously, the `hwlatdetect` tool that the Cloud-native Functions (CNF) tests image uses for running latency tests on a CNF-enabled {product-title} cluster was configured with a detection period of `10` seconds. This configuration when combined with the detection width configuration of `0.95` seconds increased the likelihood of `hwlatdetect` missing a latency spike, because the tool monitors a node about 9.5 percent of the time over the allocated detection period. With this update, the detection width is set to `1` second, so that the tool can now monitor nodes for about 95 percent of the time over the allocated detection period. The remaining 5 percent of monitoring time is left unallocated so that the kernel can perform system tasks. (link:https://issues.redhat.com/browse/OCPBUGS-12433[*OCPBUGS-12433*])

Reviewer: I am having a hard time understanding this bullet. I think it could be the first sentence that is tripping me up. I am having a hard time suggesting a rewrite because I am not sure if I am messing with any of  the terminology.

Reviewer: Percent should be the symbol % based on isg https://www.ibm.com/docs/en/ibm-style?topic=measurement-numbers#proportions

Reviewer: It also might be worth checking but you have the time in backticks and I am not 100% sure on if they need to be or not.

Author: Yeah, these numerical-based notes are hard to write. I'll need to pull in a specialized writer as I'm not familiar with this component.

Author: Noun string is correct. See these [docs](https://docs.openshift.com/container-platform/4.12/scalability_and_performance/cnf-performing-platform-verification-latency-tests.html).
